x-pack/plugins/alerting/common/routes/maintenance_window/apis/find/schemas/v1.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { maintenanceWindowResponseSchemaV1 } from '../../../response';
+
+export const findMaintenanceWindowsRequestQuerySchema = schema.object({
```

Comment:
Let's add a validation on the total number of documents that can be returned like:

```
const MAX_DOCS_PER_PAGE = 10000

export const findMaintenanceWindowsRequestQuerySchema = schema.object(
  { // omited fro brevety },
  {
    validate: (params) => {
      const pageAsNumber = params.page ?? 0;
      const perPageAsNumber = params.per_page ?? 0;

      if (Math.max(pageAsNumber, pageAsNumber * perPageAsNumber) > MAX_DOCS_PER_PAGE) {
        return `The number of documents is too high. Paginating through more than ${MAX_DOCS_PER_PAGE} documents is not possible.`;
      }
    },
  }
)
```

Reply:
Reason https://github.com/elastic/kibana/pull/197172#issuecomment-2437087034 🙂

Reply:
As I understood schema will validate `min` and max` by it self. Do we need additional validation here to have more understandable error message?

Reply:
It is not enough! I can set the page to `9999` and the `perPage` to `99` which is way over the 10K documents. This is why we need the extra validation especially the `pageAsNumber * perPageAsNumber`. Maybe we do not need to `Math.max(pageAsNumber, pageAsNumber * perPageAsNumber) > MAX_DOCS_PER_PAGE` and only do `pageAsNumber * perPageAsNumber > MAX_DOCS_PER_PAGE` as the `max` on `page`'s schema will validate that.

Reply:
You are right! Good point!